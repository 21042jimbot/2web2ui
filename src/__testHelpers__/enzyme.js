export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
