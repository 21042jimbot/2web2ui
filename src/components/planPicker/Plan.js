import React from 'react';
import _ from 'lodash';
import PropTypes from 'prop-types';
import { getPlanPrice } from 'src/helpers/billing';
import styles from './PlanPicker.module.scss';

class Plan extends React.Component {
  getPrice(plan) {
    const priceInfo = getPlanPrice(plan);

    return priceInfo.price
      ? <span><strong>${priceInfo.price.toLocaleString()}</strong>/{priceInfo.intervalShort}</span>
      : <strong>Free</strong>;
  }

  render() {
    const { plan, className, ...rest } = this.props;
    const overage = plan.isFree
      ? 'Full-featured developer account'
      : `$${plan.overage.toFixed(2)}/ thousand extra emails`;

    const ip = plan.includesIp
      ? ', first dedicated IP address is free'
      : '';

    return (
      <a className={className} {...rest} >
        <span
          className={styles.MainLabel}><strong>{ plan.volume.toLocaleString() }</strong> emails/month for {this.getPrice(plan)}</span>
        <span className={styles.SupportLabel}>{ overage }{ ip }</span>
      </a>
    );
  }
}

function eitherMonthlyOrHourly(props, propName, componentName) {
  const hasMonthly = _.has(props, 'monthly');
  const hasHourly = _.has(props, 'hourly');

  // If both are provided, or neither, error
  if ((hasMonthly && hasHourly) || !(hasMonthly || hasHourly)) {
    return new Error('Plan\'s pricing should either be hourly or monthly but not both');
  }
}

Plan.propTypes = {
  plan: PropTypes.shape({
    volume: PropTypes.number.isRequired,
    monthly: eitherMonthlyOrHourly,
    hourly: eitherMonthlyOrHourly,
    overage: PropTypes.number.isRequired,
    includesIp: PropTypes.bool,
    isFree: PropTypes.bool
  }).isRequired
};

export default Plan;
