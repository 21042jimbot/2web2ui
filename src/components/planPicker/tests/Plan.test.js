import { shallow } from 'enzyme';
import React from 'react';
import Plan from '../Plan';

jest.mock('src/hooks/useHibanaOverride', () => styles => styles);

describe('Plan:', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      plan: {
        monthly: 100,
        overage: 0.9,
        volume: 200,
      },
    };

    wrapper = shallow(<Plan {...props} />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders free and ip correctly', () => {
    const plan = {
      monthly: 0,
      isFree: true,
      overage: 0.8,
      includesIp: true,
      volume: 300,
    };
    wrapper.setProps({ plan });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders hourly plan correctly', () => {
    const plan = {
      hourly: 0.22,
      overage: 0.8,
      includesIp: true,
      volume: 300,
    };
    wrapper.setProps({ plan });
    expect(wrapper).toMatchSnapshot();
  });
});
