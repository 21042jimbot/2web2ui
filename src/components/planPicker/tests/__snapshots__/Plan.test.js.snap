// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plan: renders correctly 1`] = `
<a>
  <span>
    <strong>
      200
    </strong>
     emails/month 
    <span>
      at 
      <strong>
        $
        100
      </strong>
      /
      mo
    </span>
  </span>
  <span>
    $0.90/ thousand extra emails
  </span>
</a>
`;

exports[`Plan: renders free and ip correctly 1`] = `
<a>
  <span>
    <strong>
      300
    </strong>
     emails/month 
    <strong>
      for Free
    </strong>
  </span>
  <span>
    Full-featured developer account
    , first dedicated IP address is free
  </span>
</a>
`;

exports[`Plan: renders hourly plan correctly 1`] = `
<a>
  <span>
    <strong>
      300
    </strong>
     emails/month 
    <span>
      at 
      <strong>
        $
        0.22
      </strong>
      /
      hr
    </span>
  </span>
  <span>
    $0.80/ thousand extra emails
    , first dedicated IP address is free
  </span>
</a>
`;
